### PR TITLE
refactor: Remove if-else block

### PR DESCRIPTION
**Pull request by robbytaylor**
I've refactored the `QueryBuilder::runSelect()` method to remove the if-else conditional.

I'm not sure if this is the best approach as I don't really like method names hidden in variables. But it also removes duplication in either condition.

### CODE REVIEW

**Comment by willtj** [comment]
Yeah I agree, I'm not too keen on method names in variables. What have you got against `if-else`? If anything I'd usually prefer an `if-else` over a ternary operator for clarity. If it's just the duplicated `$this->toCompiledQuery()` then does this do the trick?

```PHP
    protected function runSelect()
    {
        $compiledQuery = $this->toCompiledQuery();

        if ($this->shouldUseScroll()){
            $this->rawResponse = $this->connection->scrollSelect($compiledQuery);
        }
        else {
            $this->rawResponse = $this->connection->select($compiledQuery);
        }

        return $this->rawResponse;
    }
```

**Comment by robbytaylor** [comment]
Everything I've read suggests avoid `if-else` for clarity. Ternary operators or early returns are usually recommended over `if-else` to improve readability. I think it's partly to reduce code indentation.

e.g. https://laracasts.com/series/simple-rules-for-simpler-code/episodes/2

With your solution there's still duplication of `$this->rawResponse = $this->connection->`

**Comment by willtj** [comment]
I get (and agree with) the reasons for avoiding `else` but I've never seen anything about avoiding `if-else`, or preferring ternary operators over `if-else` - I'd definitely say that `if-else` is clearer to read and results in fewer mistakes.

I also don't think the duplication of `$this->rawResponse = $this->connection->` is a problem - if it's clear as it is and isn't making the code less efficient is there any reason to change it?

**Comment by alasdairmackenzie** [comment]
Completely agree on your comments about `if-else` in that it tends to increase complexity and reduce readability.  In this case I'm not sure either way has an advantage as there's still a decision happening when selecting the strategy to use.

For the sake of argument, you could also do it the following way (which I don't recommend due to line length).  My point being that I think we're debating coding style rather than complexity.   And IMO both are equally as readable as one another.

```php
protected function runSelect()
{
    $compiledQuery = $this->toCompiledQuery();
    $shouldUseScroll = $this->shouldUseScroll();
    $connection =  $this->connection;

    $this->rawResponse = $shouldUseScroll ? $connection->scrollSelect($compiledQuery) : $connection->select($compiledQuery);

    return $this->rawResponse;
 }
```

**Comment by alasdairmackenzie** [comment]
TL;DR I think we should read it as it is

**Comment by robbytaylor** [comment]
No worries, I think I was just getting carried away as I was doing https://github.com/designmynight/laravel-elasticsearch/pull/8
